core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -241,12 +245,17 @@ public RemoteLogManager(RemoteLogManagerConfig rlmConfig,
 
         indexCache = new RemoteIndexCache(rlmConfig.remoteLogIndexFileCacheTotalSizeBytes(), remoteLogStorageManager, logDir);
         delayInMs = rlmConfig.remoteLogManagerTaskIntervalMs();
-        rlmScheduledThreadPool = new RLMScheduledThreadPool(rlmConfig.remoteLogManagerThreadPoolSize());
+        rlmCopyThreadPool = new RLMScheduledThreadPool(rlmConfig.remoteLogManagerThreadPoolSize(),
+            "RLMCopyThreadPool", "kafka-rlm-copy-thread-pool-");
+        rlmExpirationThreadPool = new RLMScheduledThreadPool(rlmConfig.remoteLogManagerThreadPoolSize(),
+            "RLMExpirationThreadPool", "kafka-rlm-expiration-thread-pool-");
+        followerThreadPool = new RLMScheduledThreadPool(rlmConfig.remoteLogManagerThreadPoolSize(),
+            "RLMFollowerScheduledThreadPool", "kafka-rlm-follower-thread-pool-");
```

Comment:
The amount of threads configured by the user and instantiated will be 2X, Why do we need a separate follower thread pool? 

We can have three different tasks for clarity: RLMCopyTask, RLMExpirationTask and RLMFollowerTask. RLMCopyTask and RLMFollowerTask can reuse the same thread-pool. WDYT?

Reply:
I think it is better to have separate thread pools for the different tasks. It will prevent the starvation of copy/follower tasks. Once we have a separate config for the follower thread pool, we can keep the threadpool small.